lib/ddtrace/tagging/metadata.rb

```diff
@@ -46,7 +46,7 @@ def set_tag(key, value = nil)
         if value.is_a?(Numeric) && !(value.is_a?(Integer) && !NUMERIC_TAG_SIZE_RANGE.cover?(value))
           set_metric(key, value)
         else
-          meta[key.to_s] = value.to_s
+          meta[key.to_s] = Core::Utils.utf8_encode(value)
```

Comment:
Should we do this to the key as well? Maybe we don't have to, but maybe we do?

Reply:
I added the key as well.